src/main/java/cat/udl/eps/softarch/demo/domain/Shelter.java

```diff
@@ -0,0 +1,44 @@
+package cat.udl.eps.softarch.demo.domain;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import lombok.Data;
+import jakarta.persistence.Id;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jdk.jfr.BooleanFlag;
+import lombok.EqualsAndHashCode;
+import org.hibernate.validator.constraints.Length;
+
+import java.time.LocalDateTime;
+
+@Entity
+@Data
+@EqualsAndHashCode(callSuper = false)
+public class Shelter extends UriEntity<String> {

```

Comment:
Try to extend UriEntity<Long> instead of String. It's better to have numeric identifiers and it will make things easier in the future

Reply:
Done. Thanks!